### PR TITLE
Add UTs for WSGI and Fix Form Handling

### PR DESCRIPTION
**Pull request by ericb-granular**
@UberMouse Ran into a few issues with WSGI JSON bodies. I finally just went ahead and implemented UTs for WSGI middleware to prevent these battles in the future.

### CODE REVIEW

File: python2/raygun4py/http_utilities.py [hunk 1]
```diff
@@ -15,12 +16,16 @@ def build_wsgi_compliant_request(request):
     if http_host is not None:
         http_host = http_host.replace(' ', '')
 
-    # TODO, better understand how to "properly" fallback to wsgi.input stream
-    http_form = request.get('form') or request.get('wsgi.input')
-    if http_form is not None:
-        http_form = dict(http_form)
```
Comment (ericb-granular) [inline python2/raygun4py/http_utilities.py]: This was the major problem - and now changed it to be much more defensive

File: python2/raygun4py/http_utilities.py [hunk 2]
```diff
@@ -15,12 +16,16 @@ def build_wsgi_compliant_request(request):
...
-
     try:
+        http_form = getattr(request, 'form', None) or request.get('form')
+
+        # fallback to wsgi.input
+        if http_form is None and 'wsgi.input' in request:
+            # we can assume WSGI keys inside this block
+            content_length = int(request.get('CONTENT_LENGTH', 0))
+            if content_length:
+                http_form = request['wsgi.input'].read(content_length)
```
Comment (slin63) [inline python2/raygun4py/http_utilities.py]: So http_form will just contain the content length?

Comment (ericb-granular) [reply python2/raygun4py/http_utilities.py]: @slin63 The string of the content `read(content_length)` will turn the binary stream into a string

Comment (slin63) [reply python2/raygun4py/http_utilities.py]: 👍

**Comment by UberMouse** [comment]
I don't know about the WSGI request model to know if this an improvement or not, but I'll trust it is :)

Thanks for updating both the Py2 and Py3 versions. We really need to do some work to unify those...

**Pull request merged**